sdk/trace/span.go

```diff
@@ -36,6 +36,43 @@ const (
 	errorEventName  = "error"
 )
 
+// ReadOnlySpan allows reading information from the data structure underlying a
+// trace.Span. It is used in places where reading information from a span is
+// necessary but changing the span isn't necessary or allowed.
+// TODO: SpanContext(), Tracer() and IsRecording() are already required by
+// trace.Span. Does it make sense to require them here, too?
```

Comment:
I think based on the separation of concerns between the API and the SDK this redundancy makes sense. The readable parts of the span could be abstracted into an internal interface shared between the two parts, but it would couple these concerns and not provide a definite interface definition to the user.

If we omitted these methods here we there could be situations where the user would need to assert that a value of this interface also satisfied the API's span which would not be ideal.

I think having these defined here makes things explicit, enforces the the separation of concerns between the API and SDK, and provides a unified useful interface to the user.

The only issue I see is if the methods from the API and SDK start to differ in signature but keep the same name. I don't think there is a way around this given the requirement that the API and SDK be separate though.

Reply:
Thanks for the feedback :+1: This is how I see this, too.